python/samples/agentchat_chess_game/main.py

```diff
@@ -0,0 +1,137 @@
+    # Current player color.
+    player_color = "white" if board.turn == chess.WHITE else "black"
+    legal_moves = ", ".join([move.uci() for move in board.legal_moves])
+    board_display = board.unicode(borders=True)
+    if last_move is None:
+        prompt = f"New Game!\nBoard:\n{board_display}\nYou play {player_color}\nYour legal moves: {legal_moves}\n"
+    prompt = f"Board:\n{board_display}\nYou play {player_color}\nAI's last move: {last_move}\nYour legal moves: {legal_moves}\n"
+    return prompt + "Enter your move in UCI format: "
+
+
+def extract_move(response: str) -> str:
+    start = response.find("<move>") + len("<move>")
+    end = response.find("</move>")
+    if start == -1 or end == -1:
+        raise ValueError("Invalid response format.")
+    return response[start:end]
+
+
+async def get_ai_move(board: chess.Board, player: AssistantAgent, max_tries: int) -> str:
+    task = get_ai_prompt(board)
+    count = 0
+    while count < max_tries:
+        result = await Console(player.run_stream(task=task))
+        count += 1
+        response = result.messages[-1].content
+        assert isinstance(response, str)
+        # Check if the response is a valid UC move.
+        try:
+            move = chess.Move.from_uci(extract_move(response))
+        except (ValueError, IndexError):
+            task = "Invalid format. Please read instruction.\n" + get_ai_prompt(board)
+            continue
+        # Check if the move is legal.
+        if move not in board.legal_moves:
+            task = "Invalid move. Please enter a move from the list of legal moves.\n" + get_ai_prompt(board)
+            continue
+        return move.uci()
+    # If the player does not provide a valid move, return a random move.
+    return get_random_move(board)
+
+
+async def main(human_player: bool, max_tries: int) -> None:
+    board = chess.Board()
+    player = create_ai_player()
+    while not board.is_game_over():
+        # Get the AI's move.
+        ai_move = await get_ai_move(board, player, max_tries)
+        # Make the AI's move.
+        board.push(chess.Move.from_uci(ai_move))
+        # Check if the game is over.
+        if board.is_game_over():
+            break
+        # Get the user's move.
+        if human_player:
+            user_move = input(get_user_prompt(board))
+        else:
+            user_move = get_random_move(board)
+        # Make the user's move.
+        board.push(chess.Move.from_uci(user_move))
+        print("--------- User --------")
+        print(user_move)
+        print("-------- Board --------")
+        print(board.unicode(borders=True))
+
+    result = "AI wins!" if board.result() == "1-0" else "User wins!" if board.result() == "0-1" else "Draw!"
+    print("----------------")
+    print(f"Game over! Result: {result}")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--human", action="store_true", help="Enable human vs. AI mode.")
+    parser.add_argument(
```

Comment:
@ekzhu,

I took a look here, I think the console logging you have is good enough for an example. 
I thought about an actual visual UI but that will be too heavy and will more likely live on a diff repo. 

BUt this is a good idea to compare models and their behaviours! 
P.S I tried the local 8b deepseek, it rambled on and got lost completely.